### ISSUE TITLE
Allow manufacturerData and serviceData filters to requestDevice() to be Uint8Array or sequence<octet>

### ISSUE THREAD

**Issue by rajsite** [WebBluetoothCG/web-bluetooth#407]
tl;dr: It would be helpful for different embedding contexts if the requestData options (ie filters) could be completely represented as types available in JSON.

I work on a language built with asm.js / wasm that runs in the browser and am writing an interface to web bluetooth for that language.

The language I'm working with already has good facilities for creating JSON strings from language specific array and object representations. This is in contrast to robust JS object creation and tracking which is currently very labor intensive (and on the language side requires a lot of book-keeping for tracking JS object references).

I would like to use a JSON representation for creating the requestData options. My binding layer will take a JSON string for requestData options and parse it before passing to requestData.

By using just data types and structures for the requestData options that are well represented in JSON I get a lot more flexibility by avoiding JS context being needed for parsing that configuration. I just run JSON.parse and pass through to requestData. If I want a higher-level api I can implement that in the language side without needing to track with updates to the JS side as new filtering capabilities are added over time.

While I can post-process the data and for example check for manufacturerData, servicesData, and the sub dataPrefix and mask fields represented in JSON as arrays of numbers and convert them to Uint8Array representations, it would be easier if the Web Bluetooth api allowed for either representation. This is more flexible for the language users as well since I don't have to manually track new byte array represented fields over time.

**Comment by reillyeon** [comment]
Note that no implementation that I am aware of supports specifying `manufacturerData` or `serviceData` filters at this time. That section of the specification is marked unstable. The `dataPrefix` and `mask` fields are currently defined as a `BufferSource` and so should likely be changed to a new typedef:

```
typedef (BufferSource or sequence<octet>) BufferOrOctetSequence;
```

I don't see an example of this in any other specification however it seems like it would be generally useful. My only concern is whether this would encourage developers not to use ArrayBuffers in places where it is preferable for performance reasons.
